### DISCUSSION TITLE
How to pass the relative paths of uploaded files to the server side.

### DISCUSSION THREAD

**Discussion by cerulean-net** [livewire/livewire#5547] [1. Help]
Hello,

I'm currently working on a file uploading system with livewire and filepond, I really appreciate the awesome screencasts.

This is the filepond part. When a directory is dropped into the field, the console.log shows '_relativePath' property with appropriate value.

```php
<div
wire:ignore
x-init="
    FilePond.create($refs.input);
    FilePond.setOptions({
        server: {
            process: (fieldName, file, metadata, load, error, progress, abort, transfer, options) => {
                @this.upload('added_files', file, load, error, progress);
                console.log(file);
            },
            revert: (filename, load) => {
                @this.removeUpload('added_files', filename, load)
            },
        },
    });
"
>
    <input type="file" id="add_files" x-ref="input" wire:model="added_files" multiple>
    <button wire:click="save">Save</button>
</div>
```

Now, I would like to have the relative paths on the server side so that I can create the same files/directories structure as they are uploaded. Is there any way to achieve that?

Thank you.

**Comment by joshhanley** [comment]
@cerulean-net not sure if the `@this.upload` will accept any more details, but maybe you could call another method to pass the relative path details through, after the `@this.upload`, something like `@this.relativePathDetails(file._relativePath)`?


Hope this helps!

**Reply by cerulean-net** [reply]
Thank you for your help and sorry for replying late.
So far my solution is like this:

```php
x-init="
    FilePond.create($refs.input);
    let i = 0;
    FilePond.setOptions({
        server: {
            process: (fieldName, file, metadata, load, error, progress, abort, transfer, options) => {
                @this.upload('added_files', file, load, error, progress);
                @this.set('relative_paths.' + i, file['_relativePath']);
                i++;
            },
```

Still having problem when reverting though, it works.

**Reply by joshhanley** [reply]
@cerulean-net nice!

Would revert not be similar? Call a method that can remove the file from the relative_paths array?
